chain/checkpointing/sub.go

```diff
@@ -0,0 +1,506 @@
+
+	payload := "{\"jsonrpc\": \"1.0\", \"id\":\"wow\", \"method\": \"createrawtransaction\", \"params\": [[{\"txid\":\"" + c.ptxid + "\",\"vout\": " + strconv.Itoa(index) + ", \"sequence\": 4294967295}], [{\"" + newTaprootAddress + "\": \"" + fmt.Sprintf("%.2f", newValue) + "\"}, {\"data\": \"" + hex.EncodeToString(data) + "\"}]]}"
+	result := jsonRPC(payload)
+	fmt.Println(result)
+	if result == nil {
+		panic("cant create new transaction")
+	}
+
+	rawTransaction := result["result"].(string)
+
+	tx, err := hex.DecodeString(rawTransaction)
+	if err != nil {
+		panic(err)
+	}
+
+	var buf [8]byte
+	binary.LittleEndian.PutUint64(buf[:], uint64(value*100000000))
+	utxo := append(buf[:], []byte{34}...)
+	utxo = append(utxo, scriptPubkeyBytes...)
+
+	hashedTx, err := TaprootSignatureHash(tx, utxo, 0x00)
+	if err != nil {
+		panic(err)
+	}
+
+	/*
+	 * Orchestrate the signing message
+	 */
+
+	f := frost.SignTaprootWithTweak(c.config, ids, hashedTx[:], c.tweakedValue[:])
+	n := NewNetwork(ids, c.sub, c.topic)
+	handler, err := protocol.NewMultiHandler(f, []byte{1, 2, 3})
+	if err != nil {
+		panic(err)
+	}
+	c.LoopHandler(ctx, handler, n)
+	r, err := handler.Result()
+	if err != nil {
+		fmt.Println(err)
+		panic(err)
+	}
+	fmt.Println("Result :", r)
+
+	// if signing is a success we register the new value
+	merkleRoot := HashMerkleRoot(pubkey, cp)
+	c.tweakedValue = HashTweakedValue(pubkey, merkleRoot)
+	c.pubkey = pubkeyShort
+	// If new config used
+	if c.newconfig != nil {
+		c.config = c.newconfig
+		c.newconfig = nil
+	}
+
+	if idsStrings[0] == c.host.ID().String() {
+		// Only first one broadcast the transaction ?
+		// Actually all participants can broadcast the transcation. It will be the same everywhere.
+		rawtx := PrepareWitnessRawTransaction(rawTransaction, r.(taproot.Signature))
+
+		payload = "{\"jsonrpc\": \"1.0\", \"id\":\"wow\", \"method\": \"sendrawtransaction\", \"params\": [\"" + rawtx + "\"]}"
+		result = jsonRPC(payload)
+		if result["error"] != nil {
+			fmt.Println(result)
+			panic("failed to broadcast transaction")
+		}
+
+		fmt.Println(result)
+
+		/* Need to keep this to build next one */
+		newtxid := result["result"].(string)
+		fmt.Println("New Txid:", newtxid)
+		c.ptxid = newtxid
+	}
+}
+
+func (c *CheckpointingSub) orderParticipantsList() []string {
+	id := c.host.ID().String()
+	var ids []string
+
+	ids = append(ids, id)
+
+	for _, p := range c.topic.ListPeers() {
+		ids = append(ids, p.String())
+	}
+
+	sort.Strings(ids)
+
+	return ids
+}
+
+func (c *CheckpointingSub) formIDSlice(ids []string) party.IDSlice {
+	var _ids []party.ID
+	for _, p := range ids {
+		_ids = append(_ids, party.ID(p))
+	}
+
+	idsSlice := party.NewIDSlice(_ids)
+
+	return idsSlice
+}
+
+func (c *CheckpointingSub) prefundTaproot() error {
+	taprootAddress := PubkeyToTapprootAddress(c.pubkey)
+
+	payload := "{\"jsonrpc\": \"1.0\", \"id\":\"wow\", \"method\": \"sendtoaddress\", \"params\": [\"" + taprootAddress + "\", 50]}"
```

Comment:
Quick question: why are you defining the raw payload instead of creating a struct and then marshalling? It may be more readable and less prone to bugs.